### ISSUE TITLE
Teams limit: Add tier limit exceeded flags

### ISSUE THREAD

**Issue by igorlesnenko** [ParabolInc/parabol#7385]
Let's add some flags in the database and API as a separate issue, so we can use them to implement various functionality in parallel.

See #7379

AC:
- Add and expose in API `tierLimitExceededAt ` flag 
- Add and expose in API `scheduledLockAt ` flag
- `lockedAt`
